master/internal/experiment_filter.go

```diff
@@ -450,14 +450,16 @@ func (e experimentFilter) toSQL(q *bun.SelectQuery,
 // group_a.value.last -> group_a, value, last
 // group_b.value.a.last -> group_b, value.a, last .
 func parseMetricsName(str string) (string, string, string, error) {
-	template := regexp.MustCompile("[[:print:]]+\\.[[:print:]]+\\.min|max|mean|last")
-	if !template.MatchString(str) {
+	template := regexp.MustCompile(`(?P<group>[[:print:]]+?)\.(?P<name>[[:print:]]+)\.(?P<qualifier>min|max|mean|last)`)
```

Comment:
oh yeah - pull this regex out of this function so we dont need to compile it on every call